concepts/classes/about.md

```diff
@@ -0,0 +1,285 @@
+In OOP, you want to create objects (_instances_) from "templates" (_classes_) so that they include certain data and functionality.
+The data properties are called _fields_ in the OOP context, the function properties are called _methods_.
+
+JavaScript did not have classes at all before they were added to the language specification in 2015, but allowed for object-oriented programming using prototype-based inheritance.
+And even though a `class` keyword is available nowadays, JavaScript is still a _prototype-based_ language.
+
+To understand what it means to be a prototype-based language and how JavaScript actually works, we will go back to the time when there were no classes.
+
+## Prototype Syntax
+
+### Constructor Function
+
+In JavaScript, the template (class) is facilitated by a regular function.
+When a function is supposed to be used as such a template, it is called a _constructor function_ and the convention is that the function name should start with a capital letter.
+Instances (objects) are derived from the template using the `new` keyword when invoking the constructor function.
+
+```javascript
+function Car() {
+  // ...
+}
+
+const myCar = new Car();
+const yourCar = new Car();
+```
+
+It is important to note that in JavaScript, the instances and the constructor function keep a relationship to each other even after the instances were created.
+Every instance object includes a hidden, internal property referred to as `[[prototype]]` in the language specification.
+It holds a reference to the value of the `prototype` key of the constructor function.
+Yes, you read that correctly, a JavaScript function can have key/value pairs because it is also an object behind the scenes.
+
+Since 2015, `[[prototype]]` can be accessed via `Object.getPrototypeOf()`.
+Before that, it was accessible via the key `__proto__` in many environments.
+
+Do not confuse the prototype of an object (`[[prototype]]`) with the `prototype` property of the constructor function.
+
+```exercism/note
+To summarize:
+
+- Constructors in JavaScript are regular functions.
+- Constructing a new instance creates an object with a relation to its constructor called its _prototype_.
+- Functions are objects (callable objects) and therefore they can have properties.
+- The constructor's (function) `prototype` property will become the instance's _prototype_.
+```
+
+### Instance Fields
+
+Often, you want all the derived objects (instances) to include some fields and pass some initial values for those when the object is constructed.
+This can be facilitated via the [`this` keyword][mdn-this].
+Inside the constructor function, `this` represents the new object that will be created via `new`.
+`this` is automatically returned from the constructor function when it is called with `new`.
+
+That means we can add fields to the new instance by adding them to `this` in the constructor function.
+
+```javascript
+function Car(color, weight) {
```

Comment:
Not sure if it is helpful to show how this is constructed under the hood as an object:

```
function Car(color, weight) {
    let thisObj = Object.create(null);
    thisObj.color = color;
    thisObj.weight = weight;
    return thisObj;
}
```
`this` removes the need to explicitly set the object and return it. Might be overkill or not needed :-)

Reply:
Since the student might not know about `Object.create` at this point, I felt this code is more confusing then helping. Imo the important part is that you don't have to explicitly return `this` and that is covered in the text above.

> `this` is automatically returned from the constructor function ...

Reply:
Sounds great @junedev. This concept looks awesome and enjoyed reading the code when reviewing.

Reply:
(fun fact, returning a _new_ object from the constructor function is in fact a way to get closures/private variables _and_ private functions from a constructor, however, you need to set the prototype at `Object.create` if you want to keep inheritance to work)